app/apollo/resolvers/subscription.js

```diff
@@ -328,13 +330,41 @@ const subscriptionResolvers = {
         if( clusterId ) validateString( 'clusterId', clusterId );
         tags.forEach( value => { validateString( 'tags', value ); } );
 
+        await validAuth(me, org_id, ACTIONS.CREATE, TYPES.SUBSCRIPTION, queryName, context, [name]);
+        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - subscription authorized`);
+
         const kubeOwnerId = await models.User.getKubeOwnerId(context);
 
         // validate the number of total subscriptions are under the limit
         await validateSubscriptionLimit( org_id, 1, context );
 
         // loads the channel
         const channel = await models.Channel.findOne({ org_id, uuid: channel_uuid });
+        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - found: ${!!channel}`);
+
+        const channelIdentifiers = channel ? [channel_uuid, channel.name] : [channel_uuid];
+        await validAuth(me, org_id, ACTIONS.READ, TYPES.CHANNEL, queryName, context, channelIdentifiers);
+        logger.info({req_id, user, org_id, name, channel_uuid, version_uuid}, `${queryName} validating - channel authorized`);
+
```

Comment:
The check for groups auth and existence was inserted into the middle of the channel validation, leading to inefficiency and poor readability.  Currently the `if(!channel){` check is many lines down the code:
```
        // Validate group access
        if( allowedGroups.length < groups.length ) {
          throw new NotFoundError(context.req.t('One or more of the passed groups were not found'));
        }
        if(!channel){
          throw new NotFoundError(context.req.t('Channel uuid "{{channel_uuid}}" not found.', {'channel_uuid':channel_uuid}), context);
        }
```
- Not separating it from the `// Validate group access` code block misleads about it's purpose.
- More importantly, the check and error for missing channel should be done before querying groups and checking auth -- the group checking is completely unnecessary if the channel is not found or not auth'd.

Reply:
Resolved